### PR TITLE
Add column stats unit test for different data types

### PR DESCRIPTION
**Pull request by kewang1024**

### CODE REVIEW

File: velox/exec/tests/TableWriterColumnStatsTest.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+using namespace facebook::velox::common;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::connector;
+using namespace facebook::velox::connector::hive;
+using namespace facebook::velox::dwio::common;
+
+class TableWriterColumnStats : public HiveConnectorTestBase {};
+
+TEST_F(TableWriterColumnStats, differentTypes) {
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriterColumnStatsTest.cpp:43]: Let's test in TableWriterTest

File: velox/exec/tests/TableWriteTest.cpp [hunk 2]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
   ASSERT_EQ(obj[TableWriteTraits::kLifeSpanContextKey], "TaskWide");
 }
 
+TEST_P(AllTableWriterTest, columnStatsDataTypes) {
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2267]:
```
s/columnStatsDataTypes/columnStatsVerification/
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 3]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+  std::vector<RowVectorPtr> input;
+  input.reserve(6);
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2269]: Why reserve 6? We only have one vector?

File: velox/exec/tests/TableWriteTest.cpp [hunk 4]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+  input.push_back(makeRowVector(
+      rowType_->names(),
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2271]: Does this cover all the supported column data and stats type combinations? thanks!

File: velox/exec/tests/TableWriteTest.cpp [hunk 5]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+      std::vector<core::TypedExprPtr>{inputField},
+      "approx_distinct");
+
+  core::TypedExprPtr inputField2 =
+      std::make_shared<const core::FieldAccessTypedExpr>(VARCHAR(), "c5");
+  auto maxDataSizeCallExpr = std::make_shared<const core::CallTypedExpr>(
+      BIGINT(),
+      std::vector<core::TypedExprPtr>{inputField2},
+      "max_data_size_for_stats");
+  auto sumDataSinzeCallExpr = std::make_shared<const core::CallTypedExpr>(
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2307]: typo?

File: velox/exec/tests/TableWriteTest.cpp [hunk 6]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+  auto outputDirectory = TempDirectoryPath::create();
+
+  std::vector<FieldAccessTypedExprPtr> groupingKeys;
+  for (int i = 0; i < partitionedBy_.size(); ++i) {
+    groupingKeys.emplace_back(std::make_shared<core::FieldAccessTypedExpr>(
+        partitionTypes_.at(i), partitionedBy_.at(i)));
+  }
+
+  // aggregation node
+  core::TypedExprPtr inputField =
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2290]:
```
s/inputField/intInputField/
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 7]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+  auto minCallExpr = std::make_shared<const core::CallTypedExpr>(
+      SMALLINT(), std::vector<core::TypedExprPtr>{inputField}, "min");
+  auto maxCallExpr = std::make_shared<const core::CallTypedExpr>(
+      SMALLINT(), std::vector<core::TypedExprPtr>{inputField}, "max");
+  auto distinctCountCallExpr = std::make_shared<const core::CallTypedExpr>(
+      VARCHAR(),
+      std::vector<core::TypedExprPtr>{inputField},
+      "approx_distinct");
+
+  core::TypedExprPtr inputField2 =
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2301]:
```
s/inputField2/strInputField/
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 8]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+  auto maxDataSizeCallExpr = std::make_shared<const core::CallTypedExpr>(
+      BIGINT(),
+      std::vector<core::TypedExprPtr>{inputField2},
+      "max_data_size_for_stats");
+  auto sumDataSinzeCallExpr = std::make_shared<const core::CallTypedExpr>(
+      BIGINT(),
+      std::vector<core::TypedExprPtr>{inputField2},
+      "sum_data_size_for_stats");
+
+  std::vector<std::string> aggregateNames = {
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2312]:
```
const std::vector<std::string> aggregateNames
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 9]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+      "min",
+      "max",
+      "approx_distinct",
+      "max_data_size_for_stats",
+      "sum_data_size_for_stats"};
+  std::vector<core::AggregationNode::Aggregate> aggregates = {
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2318]:
```
const std::vector<core::AggregationNode::Aggregate> aggregates
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 10]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+      core::AggregationNode::Aggregate{minCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{maxCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{distinctCountCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{maxDataSizeCallExpr, nullptr, {}, {}},
+      core::AggregationNode::Aggregate{sumDataSinzeCallExpr, nullptr, {}, {}},
+  };
+  auto aggregationNode = std::make_shared<core::AggregationNode>(
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2325]:
```
nit: const auto aggregationNode = 
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 11]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+       makeFlatVector<int16_t>(1'000, [&](auto row) { return row; }),
+       makeFlatVector<float>(1'000, [&](auto row) { return row + 33.23; }),
+       makeFlatVector<double>(1'000, [&](auto row) { return row + 33.23; }),
+       makeFlatVector<StringView>(1'000, [&](auto row) {
+         return StringView::makeInline(fmt::format("str_{}", row * 3));
+       })}));
+  createDuckDbTable(input);
+  auto outputDirectory = TempDirectoryPath::create();
+
+  std::vector<FieldAccessTypedExprPtr> groupingKeys;
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2283]:
```
s/groupingKeys/groupingKeyFields/
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 12]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+                              rowType_->children(),
+                              partitionedBy_,
+                              nullptr,
+                              makeLocationHandle(outputDirectory->path))),
+                      false,
+                      CommitStrategy::kNoCommit)
+                  .planNode();
+
+  int columnStatsStartIndex = 3 + partitionedBy_.size();
+  RowVectorPtr result = AssertQueryBuilder(plan).copyResults(pool());
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2354]:
```
const RowVectorPtr result = ...
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 13]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+                              rowType_->names(),
+                              rowType_->children(),
+                              partitionedBy_,
+                              nullptr,
+                              makeLocationHandle(outputDirectory->path))),
+                      false,
+                      CommitStrategy::kNoCommit)
+                  .planNode();
+
+  int columnStatsStartIndex = 3 + partitionedBy_.size();
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2353]:
```
s/columnStatsStartIndex/nextColumnStatsIndex/
```

File: velox/exec/tests/TableWriteTest.cpp [hunk 14]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+                          makeHiveInsertTableHandle(
+                              rowType_->names(),
+                              rowType_->children(),
+                              partitionedBy_,
+                              nullptr,
+                              makeLocationHandle(outputDirectory->path))),
+                      false,
+                      CommitStrategy::kNoCommit)
+                  .planNode();
+
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2352]: Can you comment why columnStatsStartIndex is set to 3 + partitionedBy_.size()? Thanks!

File: velox/exec/tests/TableWriteTest.cpp [hunk 15]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+  int columnStatsStartIndex = 3 + partitionedBy_.size();
+  RowVectorPtr result = AssertQueryBuilder(plan).copyResults(pool());
+  auto minStatsVector =
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2355]: nit: add const to vectors below. thanks!

File: velox/exec/tests/TableWriteTest.cpp [hunk 16]
```diff
@@ -2262,6 +2264,114 @@ TEST_P(AllTableWriterTest, tableWriteOutputCheck) {
...
+  input.reserve(6);
+  input.push_back(makeRowVector(
+      rowType_->names(),
+      {makeFlatVector<int64_t>(1'000, [&](auto row) { return 1; }),
+       makeFlatVector<int32_t>(1'000, [&](auto row) { return 1; }),
+       makeFlatVector<int16_t>(1'000, [&](auto row) { return row; }),
+       makeFlatVector<float>(1'000, [&](auto row) { return row + 33.23; }),
+       makeFlatVector<double>(1'000, [&](auto row) { return row + 33.23; }),
+       makeFlatVector<StringView>(1'000, [&](auto row) {
+         return StringView::makeInline(fmt::format("str_{}", row * 3));
```
Comment (xiaoxmeng) [inline velox/exec/tests/TableWriteTest.cpp:2278]: nit: std::to_string(row) might be easier to verify.

**Review by xiaoxmeng — APPROVED** [review]
@kewang1024 LGTM % minors. Thanks!

File: velox/exec/tests/TableWriteTest.cpp (hunk 2 above)
Comment (kewang1024) [reply velox/exec/tests/TableWriteTest.cpp:2270]: we have column stats test underneath that verify the response data layout, this test focuses on different data types, so I think it's better to keep this name

**Review by xiaoxmeng — APPROVED** [review]
@kewang1024 thanks for the update!

**Pull request closed without merging**